src/components/main-view/main-view.jsx

```diff
@@ -27,15 +33,25 @@ export class MainView extends React.Component {
 
   onMovieClick(movie) {this.setState({selectedMovie: movie }); }
   goBack() {this.setState({selectedMovie: null}); }
+  onLoggedIn(user) {this.setState({user}); }
+  startRegister() {this.setState({register: true}); }
+  endRegister() {this.setState({register: null}); }
 
   render() {
-    const { movies, selectedMovie } = this.state;
+    const { movies, selectedMovie, user, register } = this.state;
+
+    if (!register) {
+      return <RegistrationView endRegister={() => this.endRegister()}/>
+    }
```

Comment:
Have another thought about this if statement here. What does it do?

Reply:
I was thinking that it will give some **results** after I type something on register boxes. But as I know this is wrong answer. Could you kindly advise me about this, please? :)

Reply:
With this logic you will always render a registration form, no matter what because the default state is an empty string, which is falsy. Your movieflix users, even the ones that already have been created will need to register again, to be able to reach the login screen.